test/test_core.py

```diff
@@ -200,3 +203,164 @@ def test_select_types(self, connection):
 
         row = connection.execute(sa.select(tb)).fetchone()
         assert row == (1, "Hello World!", 3.5, True, now, today)
+
+
+class TestWithClause(TablesTest):
+    run_create_tables = "each"
+
+    @staticmethod
+    def _create_table_and_get_desc(connection, metadata, **kwargs):
+        table = Table(
+            "clause_with_test",
+            metadata,
+            Column("id", types.UInt32, primary_key=True),
+            **kwargs,
+        )
+        table.create(connection)
+
+        session: ydb.Session = connection.connection.driver_connection.pool.acquire()
```

Comment:
**nitpick**: We have to return session to the pool here, but it's not crucial in tests I think

Reply:
Added calling of `delete()` method for session